create_database.py

```diff
@@ -0,0 +1,136 @@
+    lists of tuples.
+    Prepare the first list for the rows of data in the 'users table'.
+    Prepare the second list for the rows of data in the 'cars table'.
+    Return the lists.
+    """
+    user_data_rows = []
+    car_data_rows = []
+
+    for data_object in data_objects:
+        if data_object:
+            assert len(data_object) == 3, "Data object needs 3 elements."
+
+			# Prepare the data rows for the 'users table'.
+            given_name = data_object[0]
+            assert name.is_valid(given_name)
+            family_name = data_object[1]
+            assert name.is_valid(family_name)
+            id_number = user_id.create(given_name, family_name)
+            assert user_id.is_valid(id_number)
+            user_data_rows.append((id_number, given_name, family_name))
+
+            # Prepare the data rows for the 'cars table'.
+            cars = data_object[2]
+            assert car.are_valid(cars)
+            for each_car in cars:
+                car_data_rows.append((each_car, id_number))
+
+    return user_data_rows, car_data_rows
+
+def create_users_table(db_cursor, user_data_rows):
+    """Create a table of users with columns for userID, given_name,
+    and family_name. Insert the user_data_rows into the table.
+    """
+    # Create 'users table' in the given database.
+    db_cursor.execute("""
+        CREATE TABLE users(userID TEXT PRIMARY KEY,
+                           given_name TEXT,
+                           family_name TEXT)
+    """)
+
+    # Insert user data rows into the 'users table'.
+    for row in user_data_rows:
+        if row:
+            assert len(row) == 3, "Wrong amount of information in the row of data."
+
+            db_cursor.execute("""INSERT INTO users(userID, given_name, family_name)
+                              VALUES(?,?,?)""", row)
+
+def create_cars_table(db_cursor, car_data_rows):
+    """Create a table of cars with columns for car and owner. Insert
+    the car_data_rows into the table.
+    """
+    # Create 'cars table' in the given database.
+    db_cursor.execute("""
+        CREATE TABLE cars(car TEXT,
+                          owner TEXT)
+    """)
+
+    # Insert car data rows into the 'cars table'.
+    for row in car_data_rows:
+        if row:
+            assert len(row) == 2, "Wrong amount of information in the row of data."
+
+            db_cursor.execute("""INSERT INTO cars(car, owner)
+                              VALUES(?,?)""", row)
+
+if __name__ == "__main__":
+    # Create users' cars database file and connect to it.
+    DB_FILENAME = 'users_cars.sqlite' # Insert desired database filename here.
+    DATABASE = sqlite3.connect(DB_FILENAME)
+    DB_CURSOR = DATABASE.cursor()
+
+    # Extract data from text file.
+    FILENAME = 'cars.txt' # Insert data input filename here.
+    DATA_PACKAGES = read_file.get_data(FILENAME, package_size=3)
+    DATA_OBJECTS = [unpack_data(DATA_PACKAGE) for DATA_PACKAGE in DATA_PACKAGES]
```

Comment:
data, particularly bad on this line, i have no idea what this line is doing if i didn't already know the challenge